src/main/java/frc/robot/autonomous/AutonomousBuilder.java

```diff
@@ -0,0 +1,41 @@
+package frc.robot.autonomous;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.commands.PathPlannerAuto;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Robot;
+import frc.utils.auto.PathPlannerUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AutonomousBuilder {
+
+	public static Map<String, Command> getAllAutos(Robot robot) {
+		Map<String, Command> autoMap = new HashMap<>();
+		autoMap.put("M231", withResetOdometry(M231(robot)));
+		autoMap.put("Rotate", new PathPlannerAuto("Rotate"));
+		autoMap.put("Rotate 2m", new PathPlannerAuto("Rotate 2m"));
+		autoMap.put("Straight 2m", new PathPlannerAuto("Straight 2m"));
+		return autoMap;
+	}
+
+	private static Command withResetOdometry(PathPlannerAuto auto) {
+		return AutoBuilder.resetOdom(auto.getStartingPose()).andThen(auto);
```

Comment:
Wtf, how they reset idonerty? They got pose estimator inside?

Reply:
they take it from AutoBuilder, I can replace it with an AutoBuilder function if you think it's more informative

Reply:
Oh right right we give them reset function I forgot